src/lib.rs

```diff
@@ -63,6 +65,40 @@
 //! |`discriminant`|`Option<syn::Expr>`|For a variant such as `Example = 2`, the `2`|
 //! |`fields`|`Option<darling::ast::Fields<__>>`|The fields associated with the variant|
 //! |`attrs`|`Vec<syn::Attribute>`|The forwarded attributes from the passed in variant. These are controlled using the `forward_attrs` attribute.|
+//!
+//! ## Examples
+//!
+//! ### Derive Macro
+//!
+//! Here's an example-invocation of a derive macro `Foo` that parses a single top-level attribute
+//! named `bar` that accepts a string value:
+//!
+//! ```ignore
+//! #[derive(Foo)]
+//! #[foo(bar = "hello")]
+//! struct Foo {
+//! }
+//! ```
+//!
+//! Which can be parsed with `darling` like so:
+//!
+//! ```ignore
+//! use darling::FromDeriveInput;
+//!
+//! #[derive(FromDeriveInput)]
+//! struct Foo {
+//!   bar: syn::LitStr,
+//! }
+//!
+//! #[proc_macro_derive(Foo, attributes(foo))]
+//! pub fn display(item: proc_macro::TokenStream) -> proc_macro::TokenStream {
+//!   let derive_input = syn::parse_macro_input!(item as syn::DeriveInput);
+//!
+//!   let foo = Foo::from_derive_input(&derive_input).unwrap();
```

Comment:
We shouldn't use `unwrap` in an example. There's a right way to check the errors which will cause them to be emitted as compile errors; I have to look for where it's demonstrated, since it's been a while since I looked at this piece of the code.

Reply:
[Here is a better approach from `derive_builder_core`](https://github.com/colin-kiegel/rust-derive-builder/blob/master/derive_builder_core/src/lib.rs#L62-L67)

```rust
    let opts = match macro_options::Options::from_derive_input(&ast) {
        Ok(val) => val,
        Err(err) => {
            return err.write_errors();
        }
    };
```

The `write_errors()` call produces a `proc_macro2::TokenStream`, so calling `into` on it will then convert it into the `proc_macro::TokenStream` expected by the derive entry-point function.